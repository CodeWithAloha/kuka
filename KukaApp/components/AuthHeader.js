import React from 'react';
import KukaLogo from '../assets/images/kuka_logo_no_text.svg';
import { StyleService, Text, useStyleSheet } from '@ui-kitten/components';
import { View } from 'react-native';

export const AuthHeader = ({ titleText, leadText }) => {
  const styles = useStyleSheet(themedStyles);

  return (
    <View style={styles.headerContainer}>
      <KukaLogo width={103} height={81} />
      <Text category="h1" status="control" style={styles.titleText}>
        {titleText}
      </Text>
      <Text style={styles.leadText} category="s1" status="control">
        {leadText}
      </Text>
    </View>
  );
};

const themedStyles = StyleService.create({
  headerContainer: {
    padding: 24,
    minHeight: 234,
    backgroundColor: 'color-primary-default',
  },
  titleText: {
    marginTop: 16,
    fontWeight: '700',
    fontFamily: 'OpenSans-Bold',
    fontSize: 30,
  },
  leadText: {
    marginTop: 3,
    fontFamily: 'OpenSans-Regular',
    fontSize: 15,
    fontWeight: '400',
  },
});
