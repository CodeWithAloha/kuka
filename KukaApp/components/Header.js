import React from 'react';
import { StyleService, Text, useStyleSheet } from '@ui-kitten/components';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

const themedStyles = StyleService.create({
  headerContainer: {
    backgroundColor: 'color-primary-default',
    fontFamily: 'OpenSans-Bold',
    fontWeight: '700',
    fontSize: 26,
    paddingLeft: 18,
    paddingBottom: 10,
  },
});

export const Header = ({ inset, text }) => {
  const insets = useSafeAreaInsets();
  const styles = useStyleSheet(themedStyles);

  return (
    <Text
      style={{
        ...styles.headerContainer,
        paddingTop: inset ? insets.top : null,
      }}
      category="h2"
      appearance="alternative"
    >
      {text}
    </Text>
  );
};
