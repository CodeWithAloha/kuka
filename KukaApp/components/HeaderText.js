import React from 'react';
import { StyleService, Text, useStyleSheet } from '@ui-kitten/components';

const themedStyles = StyleService.create({
  headerContainer: {
    backgroundColor: 'color-primary-default',
    fontFamily: 'OpenSans-Bold',
    fontWeight: '700',
    fontSize: 26,
    paddingTop: 20,
    paddingLeft: 18,
    paddingBottom: 10,
  },
});

export const HeaderText = ({ text }) => {
  const styles = useStyleSheet(themedStyles);

  return (
    <Text style={styles.headerContainer} status="control" category="h2">
      {text}
    </Text>
  );
};
