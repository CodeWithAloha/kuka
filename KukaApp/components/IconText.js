import React from 'react';
import { View } from 'react-native';
import { Text, Icon, useTheme } from '@ui-kitten/components';

export const IconText = ({ name, children, style, ...rest }) => {
  const theme = useTheme();

  return (
    <View style={[{ flexDirection: 'row', alignItems: 'center' }, style]}>
      <Icon
        name={name}
        width={24}
        height={24}
        fill={theme['text-hint-color']}
        style={{ marginRight: 8 }}
      />
      <Text appearance="hint">{children}</Text>
    </View>
  );
};
