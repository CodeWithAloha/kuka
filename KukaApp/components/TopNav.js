import React from 'react';
import { View } from 'react-native';
import {
  TopNavigation,
  TopNavigationAction,
  Icon,
  useTheme,
  Text,
} from '@ui-kitten/components';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

const BackIcon = props => {
  const theme = useTheme();
  return <Icon {...props} name="arrow-back" fill={theme['color-white']} />;
};

const BackAction = navigation => {
  return <TopNavigationAction onPress={navigation.goBack} icon={BackIcon} />;
};

export const TopNav = ({ navigation, title }) => {
  const insets = useSafeAreaInsets();
  const theme = useTheme();

  // taking this approach because the verticalPadding in TopNavigation cannot be overriden by topPadding
  return (
    <View
      style={{
        paddingTop: insets.top,
        backgroundColor: theme['color-primary-default'],
      }}
    >
      <TopNavigation
        style={{
          backgroundColor: theme['color-primary-default'],
        }}
        accessoryLeft={() => BackAction(navigation)}
        title={title ? <Text appearance="alternative">{title}</Text> : null}
        alignment="center"
      />
    </View>
  );
};
