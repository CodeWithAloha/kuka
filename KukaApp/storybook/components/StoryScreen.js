import * as React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { SafeAreaView } from 'react-native';

const styles = {
  flex: 1,
};

export default function StoryScreen(props) {
  return (
    <SafeAreaProvider>
      <SafeAreaView style={styles}>{props.children}</SafeAreaView>
    </SafeAreaProvider>
  );
}
