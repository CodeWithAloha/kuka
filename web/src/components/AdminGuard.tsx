import type { ReactNode } from 'react';
import React from 'react';
import { Redirect } from 'react-router-dom';
import useAuth from '../hooks/useAuth';

interface AdminGuardProps {
  children?: ReactNode;
}

function AdminGuard({ children }: AdminGuardProps) {
  const { isAuthenticated, user } = useAuth();

  if (!isAuthenticated || !user.isAdmin) {
    return <Redirect to="/login" />;
  }

  return (
    <>
      {children}
    </>
  );
}

export default AdminGuard;
