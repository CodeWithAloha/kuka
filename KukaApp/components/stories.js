import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { HeaderText } from './HeaderText';
import { Layout } from '@ui-kitten/components';

storiesOf('HeaderText', module).add('with text=', () => (
  <Layout style={{ height: '100%' }}>
    <HeaderText text="Just the defaults" />
  </Layout>
));
