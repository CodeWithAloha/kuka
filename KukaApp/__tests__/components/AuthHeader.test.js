import React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import BootComponent from '../BootComponent';
import { AuthHeader } from '../../components/AuthHeader';

describe('AuthHeader', () => {
  describe('Render', () => {
    afterEach(cleanup);
    it('has title and subtitle', async () => {
      const { queryByText, toJSON } = render(
        <BootComponent>
          <AuthHeader titleText="KukaNow" leadText="My subtitle" />
        </BootComponent>
      );
      expect(queryByText('KukaNow')).toBeTruthy();
      expect(queryByText('My subtitle')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
