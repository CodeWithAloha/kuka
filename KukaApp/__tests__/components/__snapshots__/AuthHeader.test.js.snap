// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader Render has title and subtitle 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#5D1B45",
          "minHeight": 234,
          "padding": 24,
          "paddingTop": 1,
        }
      }
    >
      <svg
        data-file-name="SvgKuka_logo_no_text"
        height={81}
        width={103}
      />
      <Text
        appearance="default"
        category="h1"
        status="control"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 36,
              "fontWeight": "800",
            },
            Object {
              "fontFamily": "OpenSans-Bold",
              "fontSize": 30,
              "fontWeight": "700",
              "marginTop": 16,
            },
          ]
        }
      >
        KukaNow
      </Text>
      <Text
        appearance="default"
        category="s1"
        status="control"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 15,
              "fontWeight": "600",
            },
            Object {
              "fontFamily": "OpenSans-Regular",
              "fontSize": 15,
              "fontWeight": "400",
              "marginTop": 3,
            },
          ]
        }
      >
        My subtitle
      </Text>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
